pasha/context.py

```diff
@@ -79,16 +87,120 @@ def array_per_worker(self, shape, dtype=np.float64):
         operations, where each worker may work with its own accumulator.
 
         Args:
-            Same as array()
+            shape (int or tuple of ints): Shape of the array.
+            dtype (DTypeLike, optional): Data type of the array.
 
         Returns:
             (numpy.ndarray) Created array object.
         """
 
-        if isinstance(shape, int):
-            return self.array((self.num_workers, shape), dtype)
-        else:
-            return self.array((self.num_workers,) + tuple(shape), dtype)
+        return self.empty(self._per_worker_shape(shape), dtype=dtype)
+
+    def zeros(self, shape, dtype=np.float64):
+        """Allocate an initialized array.
+
+        Equivalent to :meth:`Context.empty`, but all elements are
+        initialized to zero.
+        """
+
+        array = self.empty(shape, dtype=dtype)
+        array[:] = 0
+
+        return array
+
+    def zeros_like(self, other):
+        """Allocate an initialized array with the same shape/dtype.
```

Comment:
The `*_like` functions in NumPy ([zeros_like docs](https://numpy.org/doc/stable/reference/generated/numpy.zeros_like.html)) let you override shape/dtype, and also express preferences for the memory layout of arrays with >1 dimension. I'm not sure how often these options are used - perhaps it's not worth implementing all of them, at least at first. I'm pretty sure I've used the dtype option, to get an array with the same shape but different dtype - but of course it's easy to call `np.zeros(a.shape, dtype=...)` as well.

Reply:
Interesting. Though that would require pasha to have its own array type, which would doubtless involve some more complexity. Part of what's so neat about your current approach is that the array objects are just regular NumPy arrays, even when you create them in shared memory.

Reply:
Nevermind that idea. I've played a bit with it, and as you suspected, it's still focused on subtypes of array, not other methods of allocating its underlying memory.

So I've updated the API to include everything from `numpy` (of course except the stuff related to the array protocol).